src/Core/WinSWCore/LogAppenders.cs

```diff
@@ -395,7 +399,7 @@ private void CopyStreamWithRotation(StreamReader reader, string extension)
                             var nextFileName = Path.Combine(baseDirectory, string.Format("{0}.{1}.#{2:D4}{3}", baseFileName, now.ToString(FilePattern), nextFileNumber, extension));
                             File.Move(logFile, nextFileName);
 
-                            writer = CreateWriter(new FileStream(logFile, FileMode.Create));
+                            writer = CreateWriter(new FileStream(logFile, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite));

```

Comment:
Why?

Reply:
That was an attempt to deal with a sharing violation that was more likely caused by the missing Dispose.